src/Microsoft.SqlTools.ServiceLayer/ShowPlan/Contracts/ExecutionPlanGraph.cs

```diff
@@ -64,17 +64,13 @@ public class ExecutionPlanNode
         public List<ExecutionPlanEdges> Edges { get; set; }
     }
 
-    public class ExecutionPlanGraphElementProperties
+    public class ExecutionPlanGraphPropertyBase
```

Comment:
I would just simplify this into a single class that takes a generic as the value `ExecutionPlanGraphProperty<T>` - that way you don't need to create the two separate implementations (which don't actually do anything different)

Reply:
(and if you wanted to have a strong type for the nested values then you should be able to just do this 

`public class NestedExecutionPlanGraphProperty extends ExecutionPlanGraphProperty<IList<ExecutionPlanGraphProperty>> { }`

Reply:
I agree with Charles on this, but I also think this class can be made an interface since the intention seems to be to facilitate creating the props list declared in `showPlanGraphUtils.cs` to contain both subclasses: 
![image](https://user-images.githubusercontent.com/87730006/149588554-9c906b63-0729-4eda-a5e4-c0423f2a64db.png)

In that case, I would rename and make `ExecutionPlanGraphPropertyBase` an interface since it's not defining any behavior and it's also not being instantiated directly and keep the 2 subclasses.